app/uk/gov/hmrc/bindingtariffclassification/repository/CaseRepository.scala

```diff
@@ -45,13 +46,42 @@ trait CaseRepository {
 
   def getByReference(reference: String): Future[Option[Case]]
 
-  def get(searchBy: CaseParamsFilter, sortedBy: Option[CaseSort]): Future[Seq[Case]]
+  def get(search: Search): Future[Seq[Case]]
 
   def deleteAll(): Future[Unit]
 }
 
 @Singleton
-class CaseMongoRepository @Inject()(mongoDbProvider: MongoDbProvider, jsonMapper: JsonObjectMapper)
+class EncryptingMongoRepository @Inject()(repository: CaseMongoRepository, crypto: Crypto, appConfig: AppConfig) extends CaseRepository {
+  private def encrypt: Case => Case = { c: Case =>
+    if (appConfig.mongoEncryption.enabled) crypto.encrypt(c) else c
+  }
+
+  private def encrypt(search: Search): Search = {
+    if (appConfig.mongoEncryption.enabled) crypto.encrypt(search) else search
+  }
+
+  private def decrypt: Case => Case = { c: Case =>
+    if (appConfig.mongoEncryption.enabled) crypto.decrypt(c) else c
+  }
+
+  override def insert(c: Case): Future[Case] = repository.insert(encrypt(c)).map(decrypt)
+
+  override def update(c: Case, upsert: Boolean): Future[Option[Case]] = repository.update(encrypt(c), upsert).map(_.map(decrypt))
+
+  override def incrementDaysElapsed(increment: Double): Future[Int] = repository.incrementDaysElapsed(increment)
+
+  override def getByReference(reference: String): Future[Option[Case]] = repository.getByReference(reference).map(_.map(decrypt))
+
+  override def get(search: Search): Future[Seq[Case]] = {
+    repository.get(encrypt(search)).map(_.map(decrypt))
+  }
+
+  override def deleteAll(): Future[Unit] = repository.deleteAll()
+}
+
+@Singleton
+class CaseMongoRepository @Inject()(mongoDbProvider: MongoDbProvider, mapper: SearchMapper)
```

Comment:
Do you think we should keep this class? Or is it adding only more work? We would need to maintain it, but we do not use it I believe

Reply:
we 100% need this class